Audit Fix: documentation updates (#17)
aave/aave-v3-horizon: docs/Horizon/Horizon-overview.md

## Overview

The Horizon Instance will introduce permissioned (RWA) assets. The Aave Pool will remain the same, except that RWA assets can be used as collateral in order to borrow stablecoins. Permissioning occurs at the asset level, with each RWA Token issuer enforcing asset-specific restrictions directly into their ERC-20 token. The Aave Pool is agnostic to each specific RWA implementation and its asset-level permissioning.

From an Issuer perspective, RwaATokens are an extension of the RWA Tokens, which are securities. These RWA-specific aTokens (which are themselves not securities) will simply signify receipt of ownership of the supplied underlying RWA Token, but holders retain control over their RWA Token and can withdraw them as desired within collateralization limits.

However, holding an RwaAToken is purposefully more restrictive than merely holding an RWA Token. RWA Tokens subject holders to Issuer whitelisting and transfer mechanisms, but RwaATokens are fully locked and cannot be transferred by holders themselves.

> To accommodate edge cases, a protocol-wide RWA aToken Transfer Admin is also added, allowing Issuers the ability to forcibly transfer RWA aTokens on behalf of end users (without needing approval). These transfers will still enforce collateralization and health factor requirements as in existing Aave peer-to-peer aToken transfers.

Reviewer: ```suggestion
For added security and robustness, a protocol-wide RWA aToken Transfer Admin is also added, allowing Issuers the ability to forcibly transfer RWA aTokens on behalf of end users (without needing approval). These transfers will still enforce collateralization and health factor requirements as in existing Aave peer-to-peer aToken transfers.
```

  - this should be considered a standard ERC20 transfer between liquidated user and liquidator, and will adhere to the same restrictions imposed by the underlying RWA Token. This scenario is described [here](#liquidation-as-a-transfer-of-underlying-rwa-token).
  - liquidators are implicitly permissioned to those already allowlisted to receive underlying RWA asset (will rely on underlying RWA asset-level permissioning imposed by RWA's `transfer` function).
    - technically any user allowlisted to hold RWA Token asset can liquidate; any further permissioning to a smaller subset of liquidators is expected to be governed off-chain.

#### Configuration

- RwaATokenManager contract address granted the RwaAToken admin role in the ACL Manager.
  - further granular RwaAToken admin permissions will be configured in the RwaATokenManager contract itself.
  - Token Issuers or relevant admin can be granted admin permissions on the RwaAToken corresponding to their specific RWA asset.
- No bridges/portals will be configured, hence no unbacked RwaATokens can be minted. 

> #### Reserve

Reviewer: ```suggestion
#### Reserve Configuration
```

> - priceFeed: different per asset, but will be required to be Chainlink-compatible

Reviewer: ```suggestion
- priceFeed: different per asset, Chainlink-compatible
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- rateStrategyParams: N/A (can be left empty)
- borrowingEnabled: false (to prevent its borrowing)
- borrowableInIsolation: false
- withSiloedBorrowing: false
- flashloanable: false
- LTV: different per asset, <100%

> - liqThreshold: different per asset, <100%
> - liqBonus: different per asset, >100%

Reviewer: ```suggestion
- liquidationThreshold: different per asset, <100%
- liquidationBonus: different per asset, >100%
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- reserveFactor: 0
- supplyCap: different per asset
- borrowCap: 0
- debtCeiling: non-zero if the RWA asset is in isolation
- liqProtocolFee: 0 (must be 0, otherwise liquidations will revert in RwaAToken due to `transferOnLiquidation`)

### Stablecoins (Borrowable Asset)

> Stablecoins can be supplied permissionlessly to earn yield. However, they will only be able to be borrowed, but disabled as collateral assets (via asset configuration, by setting Liquidation Threshold to 0). Borrowing will be implicitly permissioned because only users that have supplied RWA assets can borrow stablecoins (except in a potential case [here](#non-allowlisted-account-can-receive-rwaatokens)). Other existing functionality remains the same as in v3.3. Stablecoin assets will be listed as usual, also working in a standard way.

Reviewer: ```suggestion
Stablecoins can be supplied permissionlessly to earn yield. However, they will only be able to be borrowed, but disabled as collateral assets (via asset configuration, by setting Liquidation Threshold to 0). Borrowing will be implicitly permissioned because only users that have supplied RWA assets can borrow stablecoins (except in a potential case [here](#non-allowlisted-account-can-receive-rwaatokens)). 
All other existing functionality remains unchanged from v3.3. Stablecoin assets will be listed and operate as usual, following the standard process.
```

#### Reserve Configuration

> - priceFeed: different per asset, but will be required to be Chainlink-compatible

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- rateStrategyParams: different per asset 
- borrowingEnabled: true 
- borrowableInIsolation: true
- withSiloedBorrowing: false
- flashloanable: true (flashborrowers also will be configured)
- LTV: 0

> - liqThreshold: 0 (to disable its use as collateral)

Reviewer: ```suggestion
- liquidationThreshold: 0 (to disable its use as collateral)
```

> - liqBonus: 100% (as it won't apply for a non-collateral asset)

Reviewer: ```suggestion
- liqBonus: N/A (can be 0)
```

- reserveFactor: different per asset
- supplyCap: different per asset
- borrowCap: different per asset
- debtCeiling: 0 (only applies to isolated asset)

> - liqProtocolFee: 0 (as it won't apply for a non-collateral asset)

Reviewer: ```suggestion
- liquidationProtocolFee: 0 (as it won't apply for a non-collateral asset)
```

## Edge Cases of Note

Consider the following scenarios involving the example permissioned `RWA_1` token.

### RWA Holder Loses Private Keys to Wallet

If a user has a borrow position but loses private keys to their wallet, this position can be migrated to a new wallet by the Issuer.

#### Assumptions

> - `RWA_1_ISSUER` has been granted the role "flashborrower". The account will not pay a premium on the flashloan amount loaned.

Reviewer: ```suggestion
- `RWA_1_ISSUER` has been granted the role "FLASH_BORROWER_ROLE". The account will not pay a premium on the flashloan amount loaned.
```

- `RWA_1_ISSUER` has been granted `AUTHORIZED_TRANSFER_ROLE` in the RwaATokenManager contract for `aRWA_1`.
- `RWA_1_ISSUER` has an off-chain agreement with `RWA_1` suppliers to migrate supplier lost positions if needed.

#### Context

> 1. `ALICE` supply `100 RWA_1`, receiving `100 aRWA_1`.
> 1. `ALICE` borrow `50 USDC`.

Reviewer: ```suggestion
1. `ALICE` supplies `100 RWA_1`, receiving `100 aRWA_1`.
1. `ALICE` borrows `50 USDC`.
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

1. `ALICE` loses her wallet private key.

> `ALICE` creates a new wallet, `ALICE2`. 

Reviewer: we can move to resolution
nbd, just feels more organized

#### Resolution

1. `RWA_1_ISSUER` creates a new multisig wallet controlled by `RWA_1_ISSUER` and `ALICE2` with 1 of 2 signers (`NEW_ALICE_WALLET`) which will eventually be fully transferred to `ALICE2`.
1. `RWA_1_ISSUER` executes a "complex" flashloan for `50 USDC` by calling `Pool.flashLoan(...)`. In the flashloan callback, `RWA_1_ISSUER` will:
    - repay the `50 USDC` debt `onBehalfOf` `ALICE`.
    - execute `RwaATokenManager.transferRwaAToken` to transfer `100 aRWA_1` to `NEW_ALICE_WALLET`.
    - `RWA_1_ISSUER` opens a new borrow position from `NEW_ALICE_WALLET` for `50 USDC`.
    - `RWA_1_ISSUER` repays flashloan using newly borrowed `50 USDC`.

>     - `RWA_1_ISSUER` revokes its signing role from `NEW_ALICE_WALLET`, fully transferring ownership to `ALICE2`.

Reviewer: ```suggestion
3. `RWA_1_ISSUER` revokes its signing role from `NEW_ALICE_WALLET`, fully transferring ownership to `ALICE2`.
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

Limitations
- There may not be ample liquidity in the Horizon Pool to cover via flashloan the debt position to migrate. Under those circumstances, it is the responsibility of the Issuer to provide liquidity or resolve as needed.

### RWA Holder Becomes Sanctioned After Creating a Debt Position

If a user creates a debt position but then becomes sanctioned, their actions may need to be blocked until further resolution. Consider the following scenario.

#### Assumptions

- `RWA_1` has a `80% LTV` in Horizon.

#### Context

- `ALICE` supply `1000 RWA_1` with a value of `$1000`, receiving `1000 aRWA_1`.
- `ALICE` borrow `100 USDC`. With `80% LTV`, she could borrow `700 USDC` more.
- At this point `ALICE` becomes sanctioned.

#### Resolution

> Option 1

Author: we should prefer option 1 over 2 as it leaves the protocol in healthier shape, wdyt should I just not even mention option 2?

> - `RWA_1_ISSUER` repays `100 USDC` debt onBehalfOf `ALICE`.

Reviewer: ```suggestion
- `RWA_1_ISSUER` repays `100 USDC` debt on behalf of `ALICE`.
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

Assumptions:
- `RWA_1_ISSUER` has been granted `AUTHORIZED_TRANSFER_ROLE` in the RwaATokenManager contract for `aRWA_1`..
- `ALICE` is allowlisted to hold `RWA_1` and has supplied `100 RWA_1` to Horizon, receiving `100 aRWA_1`.
- `BOB` is not allowlisted to hold `RWA_1`.

1. `RWA_1_ISSUER` executes `authorizedTransfer` from `ALICE` to `BOB` for `100 aRWA_1`.
1. `BOB` sets `RWA_1` to be used as collateral.
1. `BOB` borrows `50 USDC` against their `aRWA_1`.

`BOB` then has created a debt position without having held any underlying RWA Token. `RWA_1_ISSUER` bears responsibility to avoid this scenario through proper execution of `authorizedTransfer`.

### `Withdraw` as a Transfer of Underlying RWA Token

By specifying an arbitrary `to` address argument in the `withdraw` function, users who have supplied RWA Tokens can withdraw them to any other allowlisted account. This should be considered a standard ERC20 transfer and will adhere to the same restrictions imposed by the underlying RWA Token. Consider the following scenario.

- `Bob` supply `100 RWA_1`.

> - `Bob` withdraw `50 RWA_1` with `to` set to `ALICE`'s wallet address.

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

  - if `ALICE` has **not** been allowlisted to hold `RWA_1`, this transaction will revert.
  - if `ALICE` has been allowlisted to hold `RWA_1`, she will receive `50 RWA_1`.

Outcome

Assuming `ALICE` has been allowlisted, the two following helpful events will be emitted:

```

> emit Withdraw(params.asset, msg.sender, params.to, amountToWithdraw);

Reviewer: better to use the signature?

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

```

> where 

Reviewer: i d use named returned params

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- `params.asset` (address) is the `RWA_1` token address.
- `msg.sender (address) is `BOB`'s account.
- `params.to` (address) is `ALICE`'s account.
- `amountToWithdraw` (uint256) is `50`, with corresponding decimals units appended (ex. if decimals is 6, this value will be `50_000_000`).

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

where 
- `from` (address) is the `RWA_1` **RwaAToken** address.

>   - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

>   - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 

Reviewer: where is the other one documented?

>   - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/4fa8ee5a28734f639d091b40c22d061505ebc02a

- `to` (address) is `ALICE`'s account.
- `value` (uint256) will match `amountToWithdraw` value.

The `RWA_1` Transfer Agent must properly record this officially as a transfer of `RWA_1` between `BOB` and `ALICE` (rather than a transfer between the `RWA_1` RwaAToken contract and `ALICE`).

### `Liquidation` as a Transfer of Underlying RWA Token

During a liquidation, collateral seized from the user being liquidated will be transferred to the liquidator. This should also be considered a standard ERC20 transfer. 

Assumptions:
- `BOB` and `ALICE` are allowlisted to hold `RWA_1`.
- `ALICE` has an off-chain legal agreement with `RWA_1_ISSUER` to be able to be a liquidator.
- `LTV` of `RWA_1` is `>80%` in Horizon.
- `RWA_1` has decimals of 8.

Consider the following scenario:

> - `BOB` supply `100 RWA_1`, and borrows `80 USDC`.

Reviewer: ```suggestion
- `BOB` supplies `100 RWA_1`, and borrows `80 USDC`.
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

> - time flies and `Bob`'s `USDC` debt grows to `120 USDC` through accumulation of interest. His position is no longer healthy and it becomes liquidatable. 

Reviewer: ```suggestion
- time flies and `BOB`'s `USDC` debt grows to `120 USDC` through accumulation of interest. His position is no longer healthy and it becomes eligible for liquidation. 
```

> - `ALICE` executes a `liquidationCall` on `BOB`'s position, and is able to earn all of `BOB`'s seized `100 RWA_1` (which includes the liquidation bonus) by repaying `BOB`'s `120 USDC` debt.

Reviewer: > includes the liquidation bonus
because it's 0 right, better to say that instead imo

Author: no for the RWA collateral there would be LB > 100%

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

Reviewer: the following part is a bit confusing to me: 

> and is able to earn all of `BOB`'s seized `100 RWA_1`

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/4fa8ee5a28734f639d091b40c22d061505ebc02a

- `ALICE` receives `100 RWA_1`.

Outcome

Two helpful events will be emitted involving the collateral asset:

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

where 
- `from` (address) is the `RWA_1` **RwaAToken** address.

>   - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 

Reviewer: same as before, this is the event from the underlying asset

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- `to` (address) is `ALICE`'s account.
- `value` (uint256) will be the liquidated `RWA_1` collateral amount, ie `10_000_000` including decimals.

```

> emit LiquidationCall(

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

  vars.actualDebtToLiquidate,
  vars.actualCollateralToLiquidate,
  msg.sender,
  params.receiveAToken
);
```

where
- `params.collateralAsset` (address) is the `RWA_1` collateral token address.
- `params.debtAsset` (address) is the `USDC` debt token address.
- `params.user` (address) is the liquidated user account, `BOB`.
- `vars.actualDebtToLiquidate` (uint256) is the amount of debt asset, `USDC`, to liquidate, with corresponding decimals units appended (ex. if decimals is 6, this value will be `120_000_000`). 
- `vars.actualCollateralToLiquidate` (uint256) is the amount of collateral asset to liquidate, which is `10_000_000 RWA_1`.
- `msg.sender` (address) is the liquidator address, ie `ALICE`'s account.
- `params.receiveAToken` (bool) will be false, as `receiveAToken` is not allowed.

The Issuer's Transfer Agent must properly record this officially as a transfer of `RWA_1` between `BOB` (liquidated user) and `ALICE` (liquidator) (rather than a transfer between the `RWA_1` RwaAToken contract and `ALICE`).

### Further Configuration

Exact configuration details for eMode, isolated mode, flash loan premiums, and liquidity mining rewards are to be determine.

## References

> - https://governance.aave.com/t/arfc-horizon-s-rwa-instance/21898

Reviewer: iirc there is a blog post about horizon. I d make sense to add it as well

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/16cff2a5c13e9451726be6c5cf417f3fc6714c8f

- reserveFactor: 0
- supplyCap: different per asset
- borrowCap: 0
- debtCeiling: non-zero if the RWA asset is in isolation
- liquidationProtocolFee: 0 (must be 0, otherwise liquidations will revert in RwaAToken due to `transferOnLiquidation`)

### Stablecoins / Permissionless Non-RWA Assets (Borrowable Asset)

Stablecoins, or other non-RWA assets, can be supplied permissionlessly to earn yield. However, they will only be able to be borrowed, but disabled as collateral assets (via asset configuration, by setting Liquidation Threshold to 0). Borrowing will be implicitly permissioned because only users that have supplied RWA assets can borrow stablecoins or other permissionless non-RWA assets (except in a potential edge case described [here](#non-allowlisted-account-can-receive-rwaatokens)). 

All other existing functionality remains unchanged from v3.3. Stablecoins, or other non-RWA assets, will be listed and operate as usual, following the standard process.

#### Reserve Configuration

- priceFeed: different per asset, Chainlink-compatible
- rateStrategyParams: different per asset 
- borrowingEnabled: true 
- borrowableInIsolation: true
- withSiloedBorrowing: false

> - flashloanable: true (flashborrowers also will be configured)

Reviewer: ```suggestion
- flashloanable: true (authorized flashborrowers can be configured)
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/4fa8ee5a28734f639d091b40c22d061505ebc02a

- liquidationBonus: N/A (can be 0)
- reserveFactor: different per asset
- supplyCap: different per asset
- borrowCap: different per asset
- debtCeiling: 0 (only applies to isolated asset)
- liquidationProtocolFee: 0 (as it won't apply for a non-collateral asset)

Stablecoins, or other non-RWA permissionless assets, may also be configured as collateral in the future, by setting `liquidationThreshold` above `0`. In such case, natively permissionless borrowing would therefore be enabled within the instance.

## Edge Cases of Note

Consider the following scenarios involving the example permissioned `RWA_1` token.

### RWA Holder Loses Private Keys to Wallet

If a user has a borrow position but loses private keys to their wallet, this position can be migrated to a new wallet by the Issuer.

#### Assumptions

- `RWA_1_ISSUER` has been granted `FLASH_BORROWER_ROLE`. The account will not pay a fee on the flashloan amount loaned.
- `RWA_1_ISSUER` has been granted `AUTHORIZED_TRANSFER_ROLE` in the RwaATokenManager contract for `aRWA_1`.
- `RWA_1_ISSUER` has an off-chain agreement with `RWA_1` suppliers to migrate supplier lost positions if needed.

#### Context

> 1. `ALICE` supplies `100 RWA_1`, receiving `100 aRWA_1`.
> 1. `ALICE` borrows `50 USDC`.
> 1. `ALICE` loses her wallet private key.

Reviewer: ```suggestion
1. `ALICE` supplies `100 RWA_1`, receiving `100 aRWA_1`.
2. `ALICE` borrows `50 USDC`.
3. `ALICE` loses her wallet private key.
```

Author: https://github.com/aave/aave-v3-horizon/pull/17/commits/4fa8ee5a28734f639d091b40c22d061505ebc02a

#### Resolution

1. `ALICE` creates a new wallet, `ALICE2`. 
1. `RWA_1_ISSUER` creates a new multisig wallet controlled by `RWA_1_ISSUER` and `ALICE2` with 1 of 2 signers (`NEW_ALICE_WALLET`) which will eventually be fully transferred to `ALICE2`.
1. `RWA_1_ISSUER` executes a "complex" flashloan for `50 USDC` by calling `Pool.flashLoan(...)`. In the flashloan callback, `RWA_1_ISSUER`:
    - repays the `50 USDC` debt `onBehalfOf` `ALICE`.
    - executes `RwaATokenManager.transferRwaAToken` to transfer `100 aRWA_1` to `NEW_ALICE_WALLET`.
    - `RWA_1_ISSUER` opens a new borrow position from `NEW_ALICE_WALLET` for `50 USDC`.
    - `RWA_1_ISSUER` repays flashloan using newly borrowed `50 USDC`.

> 1. `RWA_1_ISSUER` revokes its signing role from `NEW_ALICE_WALLET`, fully transferring ownership to `ALICE2`.

Author: was relying on the auto-conversion in numbering for .md files but i'll be more explicit 

https://github.com/aave/aave-v3-horizon/pull/17/commits/4fa8ee5a28734f639d091b40c22d061505ebc02a

    - technically any user allowlisted to hold RWA Token asset can liquidate; any further permissioning to a smaller subset of liquidators is expected to be governed off-chain.

#### Configuration

- RwaATokenManager contract address granted the RwaAToken admin role in the ACL Manager.
  - further granular RwaAToken admin permissions will be configured in the RwaATokenManager contract itself.
  - Token Issuers or relevant admin can be granted admin permissions on the RwaAToken corresponding to their specific RWA asset.
- No bridges/portals will be configured, hence no unbacked RwaATokens can be minted. 

#### Reserve Configuration

- priceFeed: different per asset, Chainlink-compatible
- rateStrategyParams: N/A (can be left empty)
- borrowingEnabled: false (to prevent its borrowing)
- borrowableInIsolation: false
- withSiloedBorrowing: false
- flashloanable: false
- LTV: different per asset, <100%
- liquidationThreshold: different per asset, <100%
- liquidationBonus: different per asset, >100%
- reserveFactor: 0
- supplyCap: different per asset

> - borrowCap: 0

Author: added clarification here: https://github.com/aave/aave-v3-horizon/pull/17/commits/4d51f968387ce34f880a32d28e70a142220cf448

  - if `ALICE` has **not** been allowlisted to hold `RWA_1`, this transaction will revert on the transfer of underlying `RWA_1`.
  - if `ALICE` has been allowlisted to hold `RWA_1`, she will receive `50 RWA_1`.

#### Outcome

Multiple events will be emitted, including two `Transfer` events - one from the underlying `RWA_1` token and one from the `aRWA_1` token being burned.

`aRWA_1` Transfer

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

From `ScaledBalanceTokenBase.sol`, where:
- `from` is the user account whose tokens are withdrawn, `BOB`.
- `to` is the zero address to signify a `burn` action.
- `value` is the amount of `aRWA_1` being burned when collateral is withdrawn, including `aToken` decimals (ie `50_000_000`).

Underlying `RWA_1` Transfer

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

From the RWA `ERC20` Token contract itself, where:
- `from` is the `RWA_1` **RwaAToken** address.
  - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 
- `to` is `ALICE`'s account.
- `value` is the amount of `RWA_1` withdrawn, including decimals (ie `50_000_000`).

> ```

Reviewer: ah, in some of them I saw the title of the contract above the event signature itself. Like above:

> Underlying `RWA_1` Transfer
> 
> ```
> event Transfer(address indexed from, address indexed to, uint256 value);
> ```

> ```

Author: I just consider the title above as a descriptor to distinguish the 2 transfer types, and details of the contract itself under. So would like to leave as is

- `ALICE` receives `100 RWA_1`.

#### Outcome

Multiple events will be emitted, including two `Transfer` events - one from the underlying `RWA_1` token and one from the `aRWA_1` token being burned.

`aRWA_1` Transfer

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

From `ScaledBalanceTokenBase.sol`, where:
- `from` is the user account being liquidated, `BOB`.
- `to` is the zero address to signify a `burn` action.
- `value` is the amount of `aRWA_1` being burned when collateral is liquidated, including decimals (ie `10_000_000_000 aRWA_1`).

Underlying `RWA_1` Transfer

```
event Transfer(address indexed from, address indexed to, uint256 value);
```

From the RWA ERC20 Token contract itself, where:
- `from` is the `RWA_1` **RwaAToken** address.
  - Note that the emitted `from` address is the **RwaAToken** smart contract rather than `BOB`'s account. 
- `to` is `ALICE`'s account.
- `value` will be the liquidated `RWA_1` collateral amount, including decimals (ie `10_000_000_000 RWA_1`).

> ```

Reviewer: add contract that belongs to
